sopel/builtins/safety.py

```diff
@@ -126,11 +126,16 @@ def setup(bot: Sopel) -> None:
 
 def safeify_url(url: str) -> str:
     """Replace bits of a URL to make it hard to browse to."""
-    parts = urlparse(url)
-    scheme = "hxx" + parts.scheme[3:]  # hxxp
-    netloc = parts.netloc.replace(".", "[.]")  # google[.]com and IPv4
-    netloc = netloc.replace(":", "[:]")  # IPv6 addresses (bad lazy method)
-    return urlunparse((scheme, netloc) + parts[2:])
+    try:
+        parts = urlparse(url)
+        scheme = parts.scheme.replace("t", "x")  # hxxp
+        netloc = parts.netloc.replace(".", "[.]")  # google[.]com and IPv4
+        netloc = netloc.replace(":", "[:]")  # IPv6 addresses (bad lazy method)
+        return urlunparse((scheme, netloc) + parts[2:])
+    except Exception:
+        pass
+    # Still try to defang URLs that fail parsing
+    return url.replace(":", "[:]").replace(".", "[.]")
```

Comment:
I know you're probably working on the tests situation, but consider not `pass`ing here:

```suggestion
        # Still try to defang URLs that fail parsing
        return url.replace(":", "[:]").replace(".", "[.]")
```